test/unit/compute_resources/ovirt_test.rb

```diff
@@ -24,4 +24,29 @@ class OvirtTest < ActiveSupport::TestCase
       assert_find_by_uuid_raises(ActiveRecord::RecordNotFound, cr)
     end
   end
+
+  describe "associating operating system" do
+    setup do
+      @operating_systems_xml = Nokogiri::XML(File.read('test/fixtures/ovirt_operating_systems.xml'))
+      @compute_resource = FactoryGirl.build(:ovirt_cr).tap do |compute_resource|
+        fog_client = mock
+        ovirt_client = OVIRT::Client.new("", "", "")
+        ovirt_client.stubs(:http_get).returns(@operating_systems_xml)
+        fog_client.stubs(:client).returns(ovirt_client)
+        compute_resource.stubs(:client).returns(fog_client)
+      end
+      skip "Not supported version of rbovirt" unless @compute_resource.supports_operating_systems?
+      @host = FactoryGirl.create(:host, :mac => 'ca:d0:e6:32:16:97')
+    end
+
+    it 'maps operating system to ovirt operating systems' do
+      @compute_resource.determine_os_type(@host).must_equal "other_linux"
```

Comment:
If you make this change (https://github.com/theforeman/foreman/pull/3308/files#r56130794) you can just call `determine_os_type(os, architecture)` without the need to create a host - which is slow